### ISSUE TITLE
Mobile app usage

### ISSUE THREAD

**Issue by UCKFA** [n3-charts/line-chart#447]
Hello again,

I have a new issue. The chart cannot be controlled on a mobile app. I have made zooming possible by adding a bar that specifies the zoom level, but panning doesn't work. It tries to take the whole page left/right. On v1, the zoom/pan were working great. Just give me a hint on how can I make it work and I will implement it.

Thank you.

**Comment by chaosmail** [comment]
Hi @UCKFA 
could you provide a Codepen to illustrate your issue and your progress on how you solved it. This would make it much easier to find a solution. Thanks a lot!
Best,
Christoph

**Comment by wasper1** [comment]
Panning/zooming doesn't work because touch* events are not applied to chart. Now it supports only mouse* events.

**Comment by lorem--ipsum** [comment]
Yeah that's because the panning/zooming components are homemade and they don't use d3's original ones (for actual reasons, I wouldn't go and implement this just out of boredom ^^). I think it doesn't require a lot of work to actually support touch events, but I don't have a tablet right now... PR's welcome !

**Comment by wasper1** [comment]
I've implemented touches. I will contribute if only i find time

**Comment by wasper1** [comment]
PR for tooltip, pan, double touch #527
